web_widget_html_markdown/static/src/js/web_widget_html_markdown.js

```diff
@@ -0,0 +1,861 @@
+                    ids.push(checklistId);
+                } else {
+                    $(this).removeAttr('id');
+                }
+            });
+            $lis.not('[id]').each(function () {
+                $(this).attr('id', 'checklist-id-' + (++max));
+            });
+        },
+        /**
+         * Allows Enter keypress in a textarea (source mode)
+         *
+         * @private
+         * @param {OdooEvent} ev
+         */
+        _onKeydown: function (ev) {
+            if (ev.which === $.ui.keyCode.ENTER) {
+                ev.stopPropagation();
+                return;
+            }
+            this._super.apply(this, arguments);
+        },
+        /**
+         * Method called when wysiwyg triggers a change.
+         *
+         * @private
+         * @param {OdooEvent} ev
+         */
+        _onReadonlyClickChecklist: function (ev) {
+            // If we are in html mode and we make a change wipe out markdown tag
+            var self = this;
+            if (ev.offsetX > 0) {
+                return;
+            }
+            ev.stopPropagation();
+            ev.preventDefault();
+            var checked = $(ev.target).hasClass('o_checked');
+            var checklistId = parseInt(($(ev.target).attr('id') || '0').replace(/^checklist-id-/, ''));
+
+            this._rpc({
+                route: '/web_editor/checklist',
+                params: {
+                    res_model: this.model,
+                    res_id: this.res_id,
+                    filename: this.name,
+                    checklistId: checklistId,
+                    checked: !checked,
+                },
+            }).then(function (value) {
+                self._setValue(value);
+            });
+        },
+        /**
+         * Method called when the wysiwyg instance is loaded.
+         *
+         * @private
+         */
+        _onLoadWysiwyg: function () {
+            var $button = this._renderTranslateButton();
+            $button.css({
+                'font-size': '15px',
+                position: 'absolute',
+                right: '+5px',
+                top: '+5px',
+            });
+            this.$el.append($button);
+        },
+        /**
+         * @private
+         * @param {OdooEvent} ev
+         */
+        _onWysiwygBlur: function (ev) {
+            ev.stopPropagation();
+            this._doAction();
+            if (ev.data.key === 'TAB') {
+                this.trigger_up('navigation_move', {
+                    direction: ev.data.shiftKey ? 'left' : 'right',
+                });
+            }
+        },
+        /**
+         * @private
+         * @param {OdooEvent} ev
+         */
+        _onWysiwygFocus: function (ev) {},
+        jsLibs: [
+            LIBS_PATH + "bootstrap-markdown.js",
+            LIBS_PATH + "showdown.js",
+            LIBS_PATH + "showdown-footnotes.js",
+            LIBS_PATH + "showdown-table.js",
+            LIBS_PATH + "showdown-toc.js",
```

Comment:
Hey, where are these libs? I don't see them in the MR.

Reply:
I also see that Odoo does not minify these JS files, so maybe it's good if we ourselves include the minified file:

https://github.com/showdownjs/showdown/tree/master/dist

Reply:
The /lib directory is in my gitignore adding.

Reply:
@thomaspaulb As i fetch the minified files, i see that in march/april showdown has exited alpha2.0.0 and released:
2..0.1,   2.0.2, 2.0,3  and 2.1 !,   lots of activity .

Including the newest version  (2.1) and testing real quick for regressions.